src/middlewared/middlewared/plugins/service_monitor.py

```diff
@@ -134,35 +144,54 @@ def run(self):
             enabled = self.isEnabled(self.name)
 
             self.logger.debug("[ServiceMonitorThread] connected=%s started=%s enabled=%s", connected, started, enabled)
+            # Everithing is OK
+            if connected and started and enabled:
+                # Do we want to reset all alerts when things get back to normal?
+                ServiceMonitorThread.reset_alerts(self.name)
+                ntries = 0
+                continue
 
-            if (connected == False):
-                self.alert("attempt %d to recover service %s\n" % (ntries + 1, self.name))
+            start_service = False
+            stop_service = False
+            ntries += 1
 
-            if (connected == True) and (started == False):
-                self.logger.debug("[ServiceMonitorThread] enabling service %s", self.name)
-                try:
-                    self.middleware.call('service.start', self.name)
-                except CallTimeout:
-                    pass
+            self.alert("attempt %d to recover service %s\n" % (ntries, self.name))
+
+            if connected:
+                if not started:
```

Comment:
This can be squashed to one line

Reply:
It can, but the whole logic of verification is incomplete IMHO, I'd rather leave it as is at the moment.
And would like the feedback of the John and Marcelo regarding it.